webapp/src/components/sidebar_buttons/sidebar_buttons.tsx

```diff
@@ -1,36 +1,45 @@
-import React from 'react';
-import {Tooltip, OverlayTrigger} from 'react-bootstrap';
+import React, { PureComponent, ReactElement } from 'react';
+import { Tooltip, OverlayTrigger } from 'react-bootstrap';
 import PropTypes from 'prop-types';
-import {makeStyleFromTheme, changeOpacity} from 'mattermost-redux/utils/theme_utils';
-
-import {RHSStates, connectUsingBrowserMessage} from 'src/constants';
-import {isDesktopApp} from 'src/utils/user_agent';
+import { makeStyleFromTheme, changeOpacity } from 'mattermost-redux/utils/theme_utils';
+
+import { RHSStates, connectUsingBrowserMessage } from 'src/constants';
+import { isDesktopApp } from 'src/utils/user_agent';
+
+import { GitLabIssuesIcon, GitLabMergeRequestIcon, GitLabReviewsIcon, GitLabTodosIcon } from './button_icons';
+import { Placement } from 'react-bootstrap/esm/Overlay';
+
+interface SidebarButtonsProps {
+    theme: any;
+    connected: boolean;
+    username: string;
+    org: string;
+    clientId: string;
+    gitlabURL: string;
+    reviews: any[];
+    todos: any[];
+    yourAssignedPrs: any[];
+    yourAssignedIssues: any[];
+    isTeamSidebar: boolean;
+    pluginServerRoute: string;
+    showRHSPlugin: () => void;
+    actions: {
+        updateRHSState: (rhsState: string) => void;
+        sendEphemeralPost: (message: string) => void;
+        getLHSData: () => Promise<void>;
+    };
+}
 
-import {GitLabIssuesIcon, GitLabMergeRequestIcon, GitLabReviewsIcon, GitLabTodosIcon} from './button_icons';
+interface SidebarButtonsState {
+    refreshing: boolean;
+}
 
-export default class SidebarButtons extends React.PureComponent {
+export default class SidebarButtons extends PureComponent<SidebarButtonsProps, SidebarButtonsState> {
     static propTypes = {
```

Comment:
We can remove this, and instead use https://www.npmjs.com/package/babel-plugin-typescript-to-proptypes, and add the plugin to the babel config file. This automatically adds the propTypes for us

Reply:
Could you please provide me an example with the possible solution, because I've tried, but I still get some errors when removing the interface and adding them in the propTypes

Reply:
We should just need to do this:

```
npm i -D babel-plugin-typescript-to-proptypes
```

Then add the babel plugin like so:
https://github.com/mattermost/mattermost-plugin-jira/blob/a4d725ccc4e898d58ebf3bb054f72f7cb30fefc6/webapp/babel.config.js#L31

Then we can remove the `static propTypes` thing altogether in the typescript file, since those will automatically be generated

Reply:
After running the command, I got this error 
<img width="556" alt="error" src="https://github.com/mattermost/mattermost-plugin-gitlab/assets/64109507/df377365-6ad1-422e-8a28-ff7131e55ca9">

And the babel plugin was already configured in the babel.config-file